src/lightning/pytorch/core/mixins/hparams_mixin.py

```diff
@@ -24,6 +26,18 @@
 _ALLOWED_CONFIG_TYPES = (AttributeDict, MutableMapping, Namespace)
 
 
+given_hyperparameters: ContextVar = ContextVar("given_hyperparameters", default=None)
```

Comment:
> Would it be correct to say that the purpose of given_hyperparameters is to bypass save_hyperparameters from saving what's on the frame from LightningModule.__init__(), because the hparams that need to be saved for instantiation is defined by the LightningCLI's config and not the LightningModule's input arguments? Is my understanding correct?

Correct.

> If we save the checkpoint using LightningCLI, would we be able to LightningModule.load_from_checkpoint in another script that doesn't use the cli?

Yes, that is the idea. `load_from_checkpoint` should work independent from cli. The only requirement would be that `jsonargparse` be installed.